demo/admin/src/products/ProductForm.tsx

```diff
@@ -68,12 +63,12 @@ type ProductFormManualFragment = Omit<GQLProductFormManualFragment, "priceList"
     datasheets: Array<GQLFinalFormFileUploadFragment>;
 };
 
-type FormValues = Omit<ProductFormManualFragment, "image" | "manufacturerCountry"> & {
+type FormValues = Omit<ProductFormManualFragment, "image"> & {
     image: BlockState<typeof rootBlocks.image>;
-    manufacturerCountry?: { id: string };
 };
 
-export function ProductForm({ id }: FormProps): React.ReactElement {
+export function ProductForm({ id, manufacturerCountry }: FormProps): React.ReactElement {
+    // TODO should we handle if fetched product.manufacturerCountry does not match value from prop? probably not
```

Comment:
If you don't check if it matches you could even not fetch it at all, and not put into form state. Currently you are not using it.

Reply:
that's right. So what could go wrong without this internal check? If the loaded entity belonged to a different value asyncSelect will become unselected and can't be reset as soon as the user clicked to open the options. 

I think it's better to remove this part of the query, because it's just complicated to keep it (especially for admin-generator)